### ISSUE TITLE
Return port as additional return value in Listen and Dial functions

### ISSUE THREAD

**Issue by rahulghangas** [renproject/aw#65]
When listening for incoming connections, specifying a port does not guarantee that it will be available. Supplying a port value of 0 picks up an available port, which can be queried as follows

```go
addr := ":0"
listener, _ := net.Listen("tcp", addr)
port := listener.Addr().(*net.TCPAddr).Port
fmt.Println("Port:", port)
```

The return value of functions `Dial` and `Listen` will change from `error` to `(uint16, error)`. Does this seem like a feature we would like? This would also solve our testing bugs implicitly and we don't have to use different ports for different tests.

**Comment by loongy** [comment]
@rahulghangas it is definitely required for the user to be able to explicitly specify a port (as it is done right now), because this is required when trying to adhere to a higher-level protocol (like RenVM) that uses a specific port.

Having said that, specifying :0 (or even defaulting to it) seems like a nice feature. Given that the port is given in the Options object — and that I wouldn't want that to change — we would just need to add a way to query the Transport for the port.

**Comment by rahulghangas** [comment]
Resolved by #80
